isis/tests/FunctionalTestsMappt.cpp

```diff
@@ -0,0 +1,173 @@
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetographicLatitude"), 9.3870849567571);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetocentricLatitude"), 9.2788326719634);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest360Longitude"), 359.14528612684);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast360Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast180Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest180Longitude"), -0.85471387315751); 
+}
+
+TEST_F(DefaultCube, FunctionalTestMapptCoordsysTest) {
+  QVector<QString> args = {"append=false", 
+                           "coordsys=userdefined", 
+                           "type=ground", 
+                           "lattype=planetographic"
+                           "londir=positivewest",
+                           "londom=180",
+                           "latitude=9.3870849567571",
+                           "longitude=0.85471387315749"};
+
+  UserInterface options(APP_XML, args);
+  Pvl appLog;
+  mappt(projTestCube, options, &appLog);
+  
+  PvlGroup mapPoint = appLog.findGroup("Results");
+
+  std::cout << mapPoint << std::endl;
+
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, mapPoint.findKeyword("FilterName"), "CLEAR");
+  EXPECT_EQ( (double) mapPoint.findKeyword("Band"), 1);
+  EXPECT_NEAR( (double) mapPoint.findKeyword("Sample"), 1, 1e-8);
+  EXPECT_NEAR( (double) mapPoint.findKeyword("Line"), 1, 1e-8);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetographicLatitude"), 9.3870849567571);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetocentricLatitude"), 9.2788326719634);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest360Longitude"), 359.14528612684);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast360Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast180Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest180Longitude"), -0.85471387315751); 
+}
+
+TEST_F(DefaultCube, FunctionalTestMapptFlatFileTest) {
+  QFile flatFile("/tmp/testOut.txt");
+  QVector<QString> args = {"to="+flatFile.fileName(), "append=false", "type=projection", "x=50000.0", "y=550000.0", "format=flat"};
+  UserInterface options(APP_XML, args);
+  Pvl appLog;
+  mappt(projTestCube, options, &appLog);
+  
+  int lineNumber = 0;
+  QTextStream flatStream(&flatFile);
+  
+  while(!flatStream.atEnd()) {
+    QString line = flatStream.readLine();
+    QStringList fields = line.split(",");
+
+    if(lineNumber == 0) {
+      EXPECT_PRED_FORMAT2(AssertQStringsEqual, fields.value(1), "Sample");
+      EXPECT_PRED_FORMAT2(AssertQStringsEqual, fields.value(2), "Line");
+    }
+    else if(lineNumber == 1) {
+      EXPECT_DOUBLE_EQ(fields.value(1).toDouble(), 1);
+      EXPECT_DOUBLE_EQ(fields.value(2).toDouble(), 1);
```

Comment:
This needs to check all of the columns of the CSV:
Sample, Line, PixelValue, X, Y, PlanetocentricLatitude, PlanetographicLatitude, PositiveEast360Longitude, PositiveEast180Longitude, PositiveWest360Longitude, PositiveWest180Longitude

You could run the app with each output once and then just check that the PVL and flatfile outputs are the same.